packages/expo-updates/src/UseUpdates.ts

```diff
@@ -64,9 +64,14 @@ export const useUpdates: () => UseUpdatesReturnType = () => {
 
   // Change the state based on native state machine context changes
   useEffect(() => {
-    getNativeStateMachineContextAsync().then((context) => {
-      setUpdatesState((updatesState) => reduceUpdatesStateFromContext(updatesState, context));
-    });
+    getNativeStateMachineContextAsync()
+      .then((context) => {
+        setUpdatesState((updatesState) => reduceUpdatesStateFromContext(updatesState, context));
+      })
+      .catch((error) => {
+        // Native call can fail (e.g. if in development mode), so catch the promise rejection and surface the error
+        setUpdatesState((updatesState) => ({ ...updatesState, checkError: error }));
```

Comment:
I think we may want to consider surfacing this in a separate property (`initializationError`) so that consumers don't need to handle both initialization errors and check errors in the `checkError` property.

Reply:
I'm concerned that the API is going to get too awkward if we add too many error properties.  Maybe what would be more useful is for `useUpdates()` to return a single property, `errors`, an array of the 5 or 10 most recent errors, e.g.

```ts
enum UseUpdatesErrorType = {
  CHECK = 'check',
  DOWNLOAD = 'download',
  OTHER = 'other'
};

type UseUpdatesReturnType {
  recentErrors: {
    type: UseUpdatesErrorType;
    error: Error;
    time: Date;
  }[];
}
```

Reply:
I think having to iterate over an array of errors each render call is more awkward (and also inefficient). 

What was previously:
```typescript
render() {
   const { initializationError, checkError, downloadError } = useUpdates();

   useEffect(() => {
     if (initializationError) {
       // do something, like setState or make a API call or something
     }
   }, [initializationError]);
}
```
would become:
```typescript
render() {
   const { errors } = useUpdates();

   const initializationError = errors.filter(e => e.type === 'OTHER' && e.message.includes('initialization')); // slow to do on every render cycle
   useEffect(() => {
     if (initializationError) {
       // do something, like setState or make a API call or something
     }
   }, [initializationError]);
}
```
which is buggy by design since the initializationError could become undefined if N more errors have been thrown since it was thrown. IMO a consumer should be able to know in perpetuity that updates failed to initialize.

This brings up a larger discussion of what the behavior of checking for updates via the JS API is in dev, and therefore what the behavior of the other fields of the state are when initialization has failed. i.e. what happens if initialization fails but the the app calls `Updates.checkForUpdateAsync` or something?

Reply:
Good points! I'll add the initializationError field.

Reply:
Done